Classes/ContextMenu/ItemProvider.php

```diff
@@ -57,6 +57,7 @@ protected function initialize()
             ->retrieveFileOrFolderObject($this->identifier);
 
         if ($resource instanceof Folder
+            && $folder->checkActionPermission('write')
```

Comment:
The variable `$folder` does not exist here. It should be `$resource`.

However, this should not be checked with folder write permissions, but if the user can modify the table `tx_falsecuredownload_folder`. So like this: `$GLOBALS['BE_USER']->check('tables_modify', 'tx_falsecuredownload_folder')`.